backend/src/middlewares/checkBuyProduct.ts

```diff
@@ -2,19 +2,20 @@ import { Response, NextFunction } from 'express';
 import {IAuthRequest} from './checkAuth';
 import {Product} from '../models/product.model';
 
-export async function checkProductAuthorizationInverted(req: IAuthRequest, res: Response, next: NextFunction) {
+export async function checkBuyProduct(req: IAuthRequest, res: Response, next: NextFunction) {
     try {
         if (!req.user) {
             console.error('need to call verifyToken before using this');
         }
         const product = await Product.findOne({
             where: {
                 id: req.params.productId,
-                UserId: req.user.userId
+                UserId: req.user.userId,
+                status: !'approved'
```

Comment:
Doesn't work this way. You should os [Op.not]

https://sequelize.org/master/manual/model-querying-basics.html#examples-with--code-op-not--code-